float64 distance
time time
